### DISCUSSION TITLE
Tab thickness

### DISCUSSION THREAD

**Discussion by dougalls** [5axes/TabAntiWarping#8] [Ideas]
Hello
I often have issues with warp pulling corners up so that they break away from the brim. This happens even when I use a very wide brim. It would be very useful to be able to add a thicker tab onto the corners of the object. Is it possible to add a customisable thickness to the tab so it could be 2 or 3 layers thick?

**Comment by 5axes** [comment]
Hello,

The goal of this plugin is just to create a geometry, the result depends of the Cura possibilities to create the support, and unfortunatly it's not so easy to create a support base, thicker than just one layer. It's not impossible but it really depends of your settings

**Comment by 5axes** [comment]
One solution ( but not really tested) could be to create in the corner a custom support ti create someting as A wall in the corner.
Support Wall Line Count = 1  or 2  
Support infile density = 0
Enable support brim = true 
![image](https://user-images.githubusercontent.com/11015345/109282501-e49fe180-781d-11eb-94e5-f4ce168a77f7.png)

**Reply by 5axes** [reply]
![image](https://user-images.githubusercontent.com/11015345/109282553-f71a1b00-781d-11eb-99d6-d1336e31afee.png)

Support Wall Line Count = 2

**Reply by dougalls** [reply]
Thanks for the response and your work on this.
I will play with these settings and see what is possible. They may affect other supports within the print which could be an issue when they are required.
Would it be possible perhaps to fill in the middle layers of the capsule shape? that would seem to be a good outcome, a solid tab well attached to the model but easy to get a scraper below.

**Reply by 5axes** [reply]
> Would it be possible perhaps to fill in the middle layers of the capsule shape? that would seem to be a good outcome, a solid tab well attached to the model but easy to get a scraper below.

Printed as a Support or as a standard part definition ?

**Reply by dougalls** [reply]
I'm not sure, I don't know enough about how the normal capsule is generated. I mean having an option for the capsule tab which fills in  the 'cup' shape generated by the current capsule option

**Reply by 5axes** [reply]
Filling the capsule with a Support will just offer the possibility to use a Infill Pattern but not several layer height. 
![image](https://user-images.githubusercontent.com/11015345/109328316-9f99a080-7859-11eb-87fe-80c80e722c8f.png)

**Reply by dougalls** [reply]
I see, I wasn't sure if the outer slope was generated as a separate wall, if it was it could possibly be generated multiple times at different radii to 'fill the cup'.

**Reply by 5axes** [reply]
Printing the Capsule as a Standard part will offer the possibility to add many initial layer as you want ... But it's difficult to remove the tab and the intersection of these parts can have a consequence on the the quality of the part .

![image](https://user-images.githubusercontent.com/11015345/109329820-577b7d80-785b-11eb-9c1f-80625b365f5c.png)

**Reply by dougalls** [reply]
I mainly face issues with large flat parts in ASA which require mechanical strength but finish is not usually so important. So trimming tabs and cleaning up is not as big a problem as warping is. How would I go about adding them as a Standard part?

**Comment by mierkismantas** [comment]
Hi, I am missing the thickness setting too. I manually changed height from 0.2 to 0.4 to get 2 layers for the tab (together with support settings to 100%) it would be more easy just to set thickness together with size and X/Y distance!

**Reply by 5axes** [reply]
Yes we could add something like that  : 
A parameter : Number of layer =>  If the number is >1 use the  **_Initial layer height_** + X * **_Layer height_**   and in this case set the **_Support Density_** to 100% 

If more user will ask for this enhancement I will modify the current release.   

 For the moment **+2.** At 5 I would reconsider the request.

**Reply by mierkismantas** [reply]
Thank you for consideration!

**Reply by dougalls** [reply]
Thanks for this.

**Reply by 5axes** [reply]
![image](https://user-images.githubusercontent.com/11015345/110770120-3a7e7b80-8259-11eb-9ba1-478ceff4b9fe.png)

**Reply by dougalls** [reply]
That looks exactly like what I need to prevent brim tearing on sharp corners. Thanks for taking the time to look at it.

**Reply by 5axes** [reply]
You can test a pre-release: [https://github.com/5axes/TabAntiWarping/releases/tag/V1.2.0](https://github.com/5axes/TabAntiWarping/releases/tag/V1.2.0)

Drag and Grop the curapackage in your cura windows for installation.

**Reply by dougalls** [reply]
I am away from my printer until the weekend but I have installed in cura and successfully placed tabs at 4 layers thick and successfully sliced. The preview looks good too. I will test print when I have access to the printer. Thanks again.

**Reply by mierkismantas** [reply]
Thanks, looking good! Suggestion: "Number of layer" => "Number of layers"/"Height in layers"

**Reply by mierkismantas** [reply]
![image](https://user-images.githubusercontent.com/78933635/110778968-aff34780-826b-11eb-8407-51ec21ea26d8.png)

**Comment by dougalls** [comment]
![20210315_072401](https://user-images.githubusercontent.com/45970192/111229464-dacbfb80-85dd-11eb-8788-987deb5505e3.jpg)
I had a chance to test, one on the left was with a 20mm brim, one on right was a 20mm brim with a 30mm tab 4 (0.2) layers thick on each corner. All other print settings the same and both on the same asa / acetone mix on the bed to help adhesion. Wasn't perfect, there is about 0.5mm of warping at the base but it is more than good enough and a huge improvement.  The tabs were easy to remove as well.
Thanks again for your work on this, it is much appreciated.

**Reply by mierkismantas** [reply]
I think i still see cracks in your model (right model, middle of the part). This probably points out that you do not have any kind of enclosure, so to get even better results i suggest using one. 
Regarding tab height, very good setting, even in enclosure long and slim abs parts still need more adhesion than a brim can offer!

**Reply by dougalls** [reply]
Yes, there are still small cracks. This was printed in an enclosure but unheated other than bed and in unheated garage so only around 35 C. I am in the process of building a heated enclosure so I can get up to 60 degrees (or wherever steppers overheat)

**Comment by pierremilcent** [comment]
Version 1.2.0 not published ? 

Thanks.

**Reply by 5axes** [reply]
On this github , not on the marketplace. Need between 7 and 10 days usualy before the validation  by Ultimaker teams
